### PR TITLE
Add more features to consider for managing the service.

### PR DESCRIPTION
**Pull request by smithago**
## Description
Update the manage service doc

## Related issues
[AB#90414](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/90414)

## Testing
NA. Only doc and generalization of Lazy stream

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,7 +12,8 @@ The Medical Imaging Server for DICOM is a .NET Core implementation of DICOMweb&t
 
 Azure Health Data Service [DICOM service](https://docs.microsoft.com/en-us/azure/healthcare-apis/dicom/deploy-dicom-services-in-azure) is a managed service and recommended for production deployment.
 
-Review [maintainance guide]( ./docs/resources/dicom-server-maintaince-guide.md) if you want to manage you own deployment. 
+Only the nuget libraries released from this repo are supported.
```
Comment (StevenBorg) [inline README.md:15]: The use of "supported" here might be misleading. We don't do official support of the OSS outputs. But I know _exactly_ what you mean (I think).  Could we reword to "Only the Nuget libraries released from this repo are meant to be used in production." (or something like that?)

File: src/Microsoft.Health.Dicom.Api/Web/DicomStreamContent.cs [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 namespace Microsoft.Health.Dicom.Api.Web;
 
-public class DicomStreamContent
+public class DicomStreamContent : IHttpStreamContent
```
Comment (StevenBorg) [inline src/Microsoft.Health.Dicom.Api/Web/DicomStreamContent.cs:11]: Will need an engineering review for these files. :-)

**Review by StevenBorg — CHANGES_REQUESTED** [review]
One quick change (maybe) for the docs.  Someone else will need to review the code.

File: src/Microsoft.Health.Dicom.Api/Web/DicomStreamContent.cs (hunk 2 above)
Comment (smithago) [reply src/Microsoft.Health.Dicom.Api/Web/DicomStreamContent.cs:11]: Yes, I will ask Will to take a quick look

File: docs/resources/dicom-server-maintaince-guide.md [hunk 3]
```diff
@@ -12,7 +12,11 @@ Below is a list of things to consider if you want to mantain your own deployment
...
+| Data redundancy | Ensure the correct SQL Database and Storage Account SKU to provide for desired data redudancy and availability.  |
 | Disaster recovery | Consider multi-region failover, back-ups and other techniques to support mission critical deployments. |
+| Privacy | Ensure up-to-date policies, tools, and resources to be compliant with Privacy. |
+| Compliance | Application Audit logs, Security scan, Data access management, Secure development life cycle, Secret management, access review... etc are some of the tools and process you will have to consider to store PHI data in HIPPA, HITRUST and ISO certified way. |
 
-> For both getting started quickly and to run production workloads at scale, we recommend the [Azure Health Data Services](https://docs.microsoft.com/en-us/azure/healthcare-apis/dicom/dicom-services-overview).
+<br>
+
+> For getting started quickly and to run production workloads at scale with 99.9 SLA and all the mentioned things managed for you, we recommend the [Azure Health Data Services](https://docs.microsoft.com/en-us/azure/healthcare-apis/dicom/dicom-services-overview).
```
Comment (wsugarman) [inline docs/resources/dicom-server-maintaince-guide.md:22]:
```suggestion
We recommend trying [Azure Health Data Services](https://azure.microsoft.com/en-us/services/health-data-services/#overview) to get up-and-running quickly for production workloads. Our managed service takes care of all of the aforementioned complexity with a guaranteed SLA. Learn more about our managed DICOM Server offering [here](https://docs.microsoft.com/en-us/azure/healthcare-apis/dicom/dicom-services-overview).
```

File: src/Microsoft.Health.Dicom.Api/Web/IHttpStreamContent.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.Health.Dicom.Api.Web;
+
+public interface IHttpStreamContent
+{
+    IEnumerable<KeyValuePair<string, IEnumerable<string>>> Headers { get; init; }
```
Comment (wsugarman) [inline src/Microsoft.Health.Dicom.Api/Web/IHttpStreamContent.cs:13]: nit: does `init` need to be defined for the interface?

File: src/Microsoft.Health.Dicom.Api/Web/IHttpStreamContent.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.Health.Dicom.Api.Web;
+
+public interface IHttpStreamContent
```
Comment (wsugarman) [inline src/Microsoft.Health.Dicom.Api/Web/IHttpStreamContent.cs:11]: Do we need an interface or is the `Stream` abstraction of the `DicomStreamContent` sufficient (eg. use `MemoryStream` sometimes)? Do you have a scenario in mind where we might want different implementations of the content itself?

File: src/Microsoft.Health.Dicom.Api/Web/DicomStreamContent.cs (hunk 2 above)
Comment (wsugarman) [inline src/Microsoft.Health.Dicom.Api/Web/DicomStreamContent.cs:11]: nit: we can probably call this `StreamContent`?

Comment (wsugarman) [reply src/Microsoft.Health.Dicom.Api/Web/DicomStreamContent.cs:11]: Or maybe something related to the fact is part of a multi-part?

**Review by wsugarman — DISMISSED** [review]

File: src/Microsoft.Health.Dicom.Api/Web/DicomStreamContent.cs (hunk 2 above)
Comment (smithago) [reply src/Microsoft.Health.Dicom.Api/Web/DicomStreamContent.cs:11]: It is http stream content with data as stream and headers as keyvalue pair

File: src/Microsoft.Health.Dicom.Api/Web/IHttpStreamContent.cs (hunk 5 above)
Comment (smithago) [reply src/Microsoft.Health.Dicom.Api/Web/IHttpStreamContent.cs:11]: to keep the Lazystream generic

**Review by StevenBorg — APPROVED** [review]
The documentation portion looks fantastic.  Thanks, Smitha!
It also looks like Will did a code review.

**Pull request merged**